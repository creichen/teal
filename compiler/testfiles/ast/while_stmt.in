fun f(a : T, b) : U = {
    while a {
	  b := b + 1;
	  a := a - 1;
    }
    return b;
}