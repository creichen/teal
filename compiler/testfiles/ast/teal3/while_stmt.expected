Module
  List
  List
    FunDecl
      IdDecl ID="f"
      Opt
        Type
          IdUse ID="U"
          List
          List
      List
        VarDecl
          IdDecl ID="a"
          Opt
            Type
              IdUse ID="T"
              List
              List
          Opt
        VarDecl
          IdDecl ID="b"
          Opt
          Opt
      Opt
        BlockStmt
          List
            WhileStmt
              Access
                IdUse ID="a"
              BlockStmt
                List
                  AssignStmt
                    Access
                      IdUse ID="b"
                    AddExpr
                      Access
                        IdUse ID="b"
                      IntConstant Value="1"
                  AssignStmt
                    Access
                      IdUse ID="a"
                    SubExpr
                      Access
                        IdUse ID="a"
                      IntConstant Value="1"
            ReturnStmt
              Access
                IdUse ID="b"
